packages/cli/src/commands/dev.js

```diff
@@ -42,6 +43,7 @@ export default {
     // Setup hooks
     nuxt.hook('watch:restart', payload => this.onWatchRestart(payload, { nuxt, builder, cmd, argv }))
     nuxt.hook('bundler:change', changedFileName => this.onBundlerChange(changedFileName))
+    nuxt.hook('build:done', showMemoryUsage)
```

Comment:
This hook is not exact as the bundle will be loaded after it. Maybe `render:resourcesLoaded` hook would fit better. or a conditional new hook before it to check `if ((this.isReady) || (this.context.options.modern && this.isModernReady))`

Reply:
Also, a timer with a low amount of timeout to ensure GC freed up build time unreferenced resources.

Reply:
Fyi, the reason I added the hook is so the memory usage text is printed before the line `Waiting for file changes`. But I indeed already noticed that when you print memory usage after that line is its 1M bigger, will have a look at this

Reply:
Ah wait, [the waiting message is also done from `build:done` hook](https://github.com/nuxt/nuxt.js/blob/e5369db2d539cf580e9ce609e895c67bc01c9b3c/packages/builder/src/builder.js#L56)...

So currently it is:
```js
ℹ Memory usage: 256 MB (RSS: 335 MB)                                                                                     09:10:16
ℹ Waiting for file changes                                                                                               09:10:16
```
but if we delay printing the memory usage it will become
```js
ℹ Waiting for file changes                                                                                               09:10:16
ℹ Memory usage: 257 MB (RSS: 335 MB)                                                                                     09:10:16
```
Allthough technically the second is more correct, personally I like the first one better. Wdyt?

Reply:
I think we should either show it in the most exact manner or not at all. Is second one delay+renderer hook or just delay?

Reply:
The second one could also be just in `commands/dev` just after `await builder.build()`

Reply:
For production it is correct but for dev, renderer will load resources in the background and final memory usage depends on its behavior (if any leak happens or big amount of data loaded)

Reply:
I checked your suggestion btw about gc but just setting a timeout doesnt seem to work.

What you see below is the result of:
```js
// packages/cli/src/commands/dev.js
    // Start Build
    await builder.build()

    // Print memory usage
    showMemoryUsage()
    for (let i = 0;i < 20; i++) {
      await new Promise(resolve => setTimeout(resolve, 100))
      showMemoryUsage()
    }

    global.gc()
    showMemoryUsage()
```
Only calling global.gc seems to have an effect

```js
ℹ Waiting for file changes                                                                                               09:21:54
ℹ Memory usage: 243 MB (RSS: 329 MB)                                                                                     09:21:54
ℹ Memory usage: 245 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:54
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 250 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 251 MB (RSS: 334 MB)                                                                                     09:21:55
ℹ Memory usage: 251 MB (RSS: 334 MB)                                                                                     09:21:56
ℹ Memory usage: 251 MB (RSS: 334 MB)                                                                                     09:21:56
ℹ Memory usage: 213 MB (RSS: 333 MB)                                                                                     09:21:56
```

Reply:
global.gc() is not exposed without a flag. We can imediately show it after renderer is ready.